llama-index-core/llama_index/core/llms/callbacks.py

```diff
@@ -236,25 +236,34 @@ def wrapper_logic(_self: Any) -> Generator[CallbackManager, None, None]:
 
             yield callback_manager
 
+        def extract_prompt(*args: Any, **kwargs: Any) -> str:
+            if len(args) > 0:
+                return str(args[0])
+            elif "prompt" in kwargs:
+                return kwargs["prompt"]
+            else:
+                return str(kwargs)
+
         async def wrapped_async_llm_predict(
             _self: Any, *args: Any, **kwargs: Any
         ) -> Any:
+            prompt = extract_prompt(*args, **kwargs)
             with wrapper_logic(_self) as callback_manager:
                 span_id = dispatcher.root.current_span_id or ""
                 model_dict = _self.to_dict()
                 model_dict.pop("api_key", None)
                 dispatcher.event(
                     LLMCompletionStartEvent(
                         model_dict=model_dict,
-                        prompt=str(args[0]),
+                        prompt=prompt,
                         additional_kwargs=kwargs,
                         span_id=span_id,
                     )
                 )
                 event_id = callback_manager.on_event_start(
                     CBEventType.LLM,
                     payload={
-                        EventPayload.PROMPT: args[0],
+                        EventPayload.PROMPT: prompt,
```

Comment:
My worry here is following:
before it was just `prompt`, now what is being `str(prompt)` not sure how it is going to affect anything, but just something to keep in mind.

Reply:
Yes I realized this too. `args[0]` should already be a str (the prompt) when present, so I figured the type casting was used to appease the linter or something like that.

Reply:
Make sense.